### PR TITLE
Improve plugins typings (Part 3)

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| License                  | MIT

See https://github.com/babel/babel/pull/14601 for more info. This PR can be reviewed by commits: each of them deal with a single plugin. It is the last one handling plugin typings.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14642"></a>

### CODE REVIEW

File: packages/babel-helpers/scripts/generate-helpers.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ export default async function generateHelpers() {
 
 import template from "@babel/template";
 
-function helper(minVersion, source) {
+function helper(minVersion: string, source: string) {
```
Comment (nicolo-ribaudo) [inline packages/babel-helpers/scripts/generate-helpers.js:17]: Isn't this file meant to run in Node.js without compiling it?

File: packages/babel-plugin-proposal-decorators/package.json [hunk 2]
```diff
@@ -34,7 +34,9 @@
     "@babel/core": "workspace:^",
     "@babel/helper-plugin-test-runner": "workspace:^",
     "@babel/traverse": "workspace:^",
+    "@types/charcodes": "^0.2.0",
     "babel-plugin-polyfill-es-shims": "^0.6.0",
+    "charcodes": "^0.2.0",
```
Comment (nicolo-ribaudo) [inline packages/babel-plugin-proposal-decorators/package.json:39]: It's also in `dependencies`. We can probably only keep it in `devDependencies`?

**Review by nicolo-ribaudo — APPROVED** [review]

File: packages/babel-helper-compilation-targets/src/utils.ts [hunk 3]
```diff
@@ -38,28 +38,35 @@ export function semverify(version: number | string): string {
...
+  const unreleasedLabel:
+    | typeof unreleasedLabels[keyof typeof unreleasedLabels]
+    | undefined =
+    // @ts-ignore unreleasedLabel is undefined when env is not safari
+    unreleasedLabels[env];
+  if (a === unreleasedLabel) {
+    return b;
+  }
+  if (b === unreleasedLabel) {
+    return a;
```
Comment (liuxingbaoyu) [inline packages/babel-helper-compilation-targets/src/utils.ts:61]: The logic here changes and doesn't always return, I'm not sure if this is expected, just a hint.

Comment (JLHwung) [reply packages/babel-helper-compilation-targets/src/utils.ts:61]: 🤔 Isn't `hasReleased === true` is equivalent to `a === unreleasedLabel || b === unreleasedLabel`?

The original logic incorrectly compared `a` with `hasUnreleased`. I revised a bit to make it more straightforward.

**Review by liuxingbaoyu — APPROVED** [review]

**Pull request merged**